pkgs/applications/editors/vscode/extensions/generate.scala

```diff
@@ -0,0 +1,185 @@
+//> using scala "3.1.1"
+//> using lib "org.http4s::http4s-ember-client:0.23.10"
+//> using lib "org.http4s::http4s-circe:0.23.10"
+//> using lib "io.circe::circe-generic:0.14.1"
+import org.http4s.client.dsl.io._
+import org.http4s.client.Client
+import org.http4s.Uri
+import org.http4s.Method.GET
+import cats.effect.IO
+import org.http4s.circe.CirceEntityCodec._
+import io.circe.generic.auto._
+import cats.effect.IOApp
+import org.http4s.ember.client.EmberClientBuilder
+import fs2.io.file.Files
+import fs2.io.file.Path
+import org.http4s.client.middleware.Logger
+import org.http4s.client.middleware.Retry
+import org.http4s.client.middleware.RetryPolicy
+import scala.concurrent.duration._
+import cats.implicits._
+import cats.data.NonEmptyList
+
+val MaxConcurrentRequests = 10
+
+def nixPrefetchExtension(
+    publisher: String,
+    name: String,
+    version: String
+): IO[String] =
+  import sys.process._
+
+  val url =
+    s"https://$publisher.gallery.vsassets.io/_apis/public/gallery/publisher/$publisher/extension/$name/$version/assetbyname/Microsoft.VisualStudio.Services.VSIXPackage"
+
+  IO.interruptibleMany(s"nix-prefetch-url $url".!!.trim)
+
+def manifest(
+    publisher: String,
+    name: String
+)(c: Client[IO]): IO[Manifest] =
+  import org.http4s.circe.CirceEntityCodec._
+
+  val url = Uri.unsafeFromString(
+    s"https://$publisher.gallery.vsassets.io/_apis/public/gallery/publisher/$publisher/extension/$name/latest/assetbyname/Microsoft.VisualStudio.Code.Manifest"
+  )
+
+  c.expect[Manifest](GET(url)).onError { case e =>
+    cats.effect.std
+      .Console[IO]
+      .errorln(
+        s"Couldn't get extension version for $publisher.$name"
+      )
+  }
+
+final case class Manifest(version: String, license: Option[String])
+
+val nixLicenses =
+  Map(
+    "MIT" -> List("mit"),
+    "Apache-2.0" -> List("asl20"),
+    "EPL-2.0" -> List("epl20"),
+    "MPL-2.0" -> List("mpl20"),
+    "BSD-3-Clause" -> List("bsd3"),
+    "Unlicense" -> List("unlicense"),
+    "MIT OR Apache-2.0" -> List("mit", "asl20"),
+    "GPL-3.0" -> List("gpl3"),
+    "GPL-3.0-or-later" -> List("gpl3Plus"),
+    "ISC" -> List("isc")
+  )
+
+case class Extension(
+    publisher: String,
+    name: String,
+    version: String,
+    sha256: String,
+    license: Option[String]
+) {
+  def metaString = license match
+    case Some(nixLicenses(knownLicenses)) =>
+      s"""
+      |  meta.license = ${knownLicenses
+        .map(l => s"lib.licenses.$l")
+        .mkString("[ ", " ", " ]")};""".stripMargin
+    case other =>
+      // side effect
+      println(
+        s"${Console.YELLOW}Unknown license for $publisher.$name: $other${Console.RESET}. Add it in overrides.nix."
+      )
```

Comment:
As the extension's manifest includes a description and the homepage of the extension it might be sensible to set `meta.homepage` and `meta.description` as well.

Reply:
It'll take a little work to render in a pretty way but I think it's worth doing. We can probably add a lot more in future PRs too.